Mathlib/NumberTheory/LSeries/Dirichlet.lean

```diff
@@ -0,0 +1,420 @@
+-/
+import Mathlib.NumberTheory.SumPrimeReciprocals
+import Mathlib.NumberTheory.VonMangoldt
+import Mathlib.NumberTheory.ZetaFunction
+import Mathlib.NumberTheory.DirichletCharacter.Bounds
+import Mathlib.NumberTheory.LSeries.Convolution
+import Mathlib.NumberTheory.LSeries.Deriv
+
+/-!
+# L-series of Dirichlet characters and arithmetic functions
+
+We collect some results on L-series of specific (arithmetic) functions, for example,
+the Möbius function `μ` or the von Mangoldt function `Λ`. In particular, we show that
+`L ↗Λ` is the negative of the logarithmic derivative of the Riemann zeta function
+on `re s > 1`; see `LSeries_vonMangoldt_eq_deriv_riemannZeta_div`.
+
+We also prove some general results on L-series associated to Dirichlet characters
+(i.e., Dirichlet L-series). For example, we show that the abscissa of absolute convergence
+equals `1` (see `DirichletCharacter.absicssaOfAbsConv`) and that the L-series does not
+vanish on the open half-plane `re s > 1` (see `DirichletCharacter.LSeries_ne_zero`).
+
+We deduce results on the Riemann zeta function (which is `L 1` or `L ↗ζ` on `re s > 1`)
+as special cases.
+
+## Tags
+
+Dirichlet L-series, Möbius function, von Mangoldt function, Riemann zeta function
+-/
+
+open scoped LSeries.notation
+
+/-- `δ` is the function underlying the arithmetic function `1`. -/
+lemma ArithmeticFunction.one_eq_delta : ↗(1 : ArithmeticFunction ℂ) = δ := by
+  ext ⟨- | _⟩ <;> simp only [map_zero, one_apply, LSeries.delta, ↓reduceIte]
+
+section Moebius
+
+/-!
+### The L-series of the Möbius function
+
+We show that `L μ s` converges absolutely if and only if `re s > 1`.
+-/
+
+namespace ArithmeticFunction
+
+open LSeries Nat Complex
+
+lemma not_LSeriesSummable_moebius_at_one : ¬ LSeriesSummable ↗μ 1 := by
+  intro h
+  refine not_summable_one_div_on_primes <| summable_ofReal.mp <| Summable.of_neg ?_
+  simp only [← Pi.neg_def, Set.indicator_comp_of_zero ofReal_zero, ofReal_inv, ofReal_nat_cast]
+  convert h.indicator {n | n.Prime} using 1
+  ext n
+  by_cases hn : n ∈ {p | p.Prime}
+  · simp only [Pi.neg_apply, Set.indicator_of_mem hn, ofReal_div, ofReal_one, ofReal_nat_cast,
+      neg_div', term_of_ne_zero hn.ne_zero, moebius_apply_prime hn, Int.reduceNeg, Int.cast_neg,
+      Int.cast_one, cpow_one]
+  · simp only [one_div, Pi.neg_apply, Set.indicator_of_not_mem hn, ofReal_zero, neg_zero]
+
+/-- The L-series of the Möbius function converges absolutely at `s` if and only if `re s > 1`. -/
+lemma LSeriesSummable_moebius_iff {s : ℂ} : LSeriesSummable ↗μ s ↔ 1 < s.re := by
+  refine ⟨fun H ↦? _, LSeriesSummable_of_bounded_of_one_lt_re (m := 1) fun n ↦ ?_⟩
+  · by_contra! h
+    have h' : s.re ≤ (1 : ℂ).re := by simp only [one_re, h]
+    exact not_LSeriesSummable_moebius_at_one <| LSeriesSummable.of_re_le_re h' H
+  · rw [abs_intCast, ← Int.cast_abs]
+    norm_cast
+    rcases eq_or_ne (μ n) 0 with h | h
+    · simp [h]
+    · rcases moebius_ne_zero_iff_eq_or.mp h with h | h <;> simp [h]
```

Comment:
This seems worth extracting; for example with
```lean
theorem moebius_apply_eq (n : ℕ) : μ n = 0 ∨ μ n = 1 ∨ μ n = -1 := by
  simp only [moebius, coe_mk]
  split_ifs
  · right
    exact neg_one_pow_eq_or ..
  · left
    rfl

theorem abs_moebius_le_one {n : ℕ} : |μ n| ≤ 1 := by
  obtain h|h|h := moebius_apply_eq n <;> simp [h]

theorem moebius_ne_zero_iff_eq_or {n : ℕ} : μ n ≠ 0 ↔ μ n = 1 ∨ μ n = -1 := by
  have := moebius_apply_eq n
  aesop
#align nat.arithmetic_function.moebius_ne_zero_iff_eq_or ArithmeticFunction.moebius_ne_zero_iff_eq_or
```

Reply:
Maybe, but I think this should be a separate PR.

Reply:
I've merged master and used `abs_moebius_le_one` to golf this proof.